### DISCUSSION TITLE
Generalized Focal Loss Head is inherited from AnchorHead?

### DISCUSSION THREAD

**Discussion by iumyx2612** [open-mmlab/mmdetection#8957] [Q&A]
the `bbox_coder` in GFLHead is 
`bbox_coder=dict(type='DistancePointBBoxCoder')` 
 
However, `anchor_generator` for GFLHead is 
```
anchor_generator=dict(
    type='AnchorGenerator',
    ratios=[1.0],
    octave_base_scale=8,
    scales_per_octave=1,
    strides=[8, 16, 32, 64, 128]),
```
Therefore, GFLHead needs a function to find center of Anchor Box generated by `anchor_generator`  
```
def anchor_center(self, anchors):
    """Get anchor centers from anchors.

    Args:
        anchors (Tensor): Anchor list with shape (N, 4), "xyxy" format.

    Returns:
        Tensor: Anchor centers with shape (N, 2), "xy" format.
    """
    anchors_cx = (anchors[..., 2] + anchors[..., 0]) / 2
    anchors_cy = (anchors[..., 3] + anchors[..., 1]) / 2
    return torch.stack([anchors_cx, anchors_cy], dim=-1)
```
Why don't MMDet implementation inherit from AnchorFreeHead and use  
`prior_generator = MlvlPointGenerator(strides)`
